### PR TITLE
Feature timeline redo

### PR DESCRIPTION
**Pull request by Dammshine**
I have not yet finished all the tasks, but I am requesting a review at this stage because will block Hindie's progress

**Tasks Completed in This PR:**
1. Refactored the socket to enforce specific types for every event received from the server or sent to the server.
2. Removed several events that are no longer used, leaving space to add these back in for future debugging if necessary.
3. Extracted socket logic from DevelopmentMode into its own dedicated hook. 
4. Moved 'tab' and 'setTab' from DevelopmentMode to GlobalStateStore.

**Planned Tasks for the Next PR:**
1. Further refactor properties passed into useSocketCommunication to GlobalStateStore. This may conflict with Hindie’s current tasks, so we will figure out the scope. 
2. Implement the actual timeline functionality.

### CODE REVIEW

File: client/src/Services/useSocketCommunication.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    };
+  }, [socketClient]);
+
+  const resetDebugSession = useCallback(() => {
+    updateNextFrame(INITIAL_BACKEND_STATE);
+    setActiveSession(false);
+    clearTypeDeclarations();
+    clearUserAnnotation();
+    setConsoleChunks([]);
+  }, [updateNextFrame, clearTypeDeclarations, clearUserAnnotation]);
```
[the remark this answers is not in the document]
Comment (Dammshine) [reply client/src/Services/useSocketCommunication.ts:73]: Good spot! The dependency isn't necessary.

File: client/src/Services/useSocketCommunication.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      resetDebugSession();
+      const file = fileSystem.getFileFromPath(currFocusFilePath);
+      if (file) {
+        console.log('Sending data:', file.data);
+        socketClient.serverAction.initializeDebugSession(file.data);
+      } else {
+        throw new Error('File not found in FS');
+      }
+    },
+    [resetDebugSession, socketClient]
```
Comment (NE1NN) [inline client/src/Services/useSocketCommunication.ts:76]: Oh yeah what about this tho? Do we really need resetDebugSession as the dependency?

Comment (Dammshine) [reply client/src/Services/useSocketCommunication.ts:76]: We don't need lol

**Review by NE1NN — APPROVED** [review]
ok lgtm

**Pull request merged**